src/DefaultBuilder/src/ConfigureWebHostBuilder.cs

```diff
@@ -161,6 +164,19 @@ public IWebHostBuilder UseSetting(string key, string? value)
             return this;
         }
 
+        private void SetWebRootPath(string value)
+        {
+            _environment.WebRootPath = Path.GetFullPath(Path.Combine(_environment.ContentRootPath, value));
+            _environment.WebRootFileProvider = new PhysicalFileProvider(_environment.WebRootPath);
+            // The StaticWebAssets loader wraps the WebRootFileProvider in order to support loading
+            // static assets from a manifest file and from a local directory. Since we modified
+            // the WebRootFileProvider, we call the StaticWebAssetsLoader to update.
+            if (_environment.IsDevelopment())
+            {
+                StaticWebAssetsLoader.UseStaticWebAssets(_environment, _configuration);
+            }
+        }
```

Comment:
Why did we stop creating a directory at WebRootPath in case it doesn't already exist? Because this could be called more than once? 

I like how this avoids redoing work if the web root path doesn't change, but recreating the PhysicalFileProvider and calling UseStaticWebAssets on each change seems unnecessary. It's probably better to capture the original WebRootPath when the WebApplicationBuilder is first constructed and defer this logic to Build() like in the earlier revision.

Reply:
> Why did we stop creating a directory at WebRootPath in case it doesn't already exist? Because this could be called more than once?

Correct. It would be called multiple times 

>  It's probably better to capture the original WebRootPath when the WebApplicationBuilder is first constructed and defer this logic to Build() like in the earlier revision.

Deferring to `Build` causes the issues outlined in https://github.com/dotnet/aspnetcore/pull/37033#discussion_r718805059. Since the `WebRootPath` is the only property on the `WebApplicationBuilder` that can be modified after the constructor runs we need some way to react to it.

All in all, I think the predictability of always setting the `WebRootFileProvider` is better than deferring to build and leaving the `HostEnvironment` in an incomplete state.